src/layers/samples/voronoi-layer/voronoi-layer.js

```diff
@@ -0,0 +1,145 @@
+// THE SOFTWARE.
+
+import {BaseLayer, assembleShader} from '../../../lib';
+import {Model, Program, Geometry} from 'luma.gl';
+const glslify = require('glslify');
+
+const DEFAULT_COLOR = [255, 0, 255];
+
+const defaultGetPosition = x => x.position;
+const defaultGetRadius = x => x.radius;
+const defaultGetColor = x => x.color || DEFAULT_COLOR;
+
+export default class VoronoiLayer extends BaseLayer {
+  /*
+   * @classdesc
+   * VoronoiLayer
+   *
+   * @class
+   * @param {object} props
+   * @param {number} props.radius - point radius in meters
+   */
+  constructor({
+    getPosition = defaultGetPosition,
+    getRadius = defaultGetRadius,
+    getColor = defaultGetColor,
+    radius = 1000,
+    ...props
+  }) {
+    super({
+      getPosition,
+      getRadius,
+      getColor,
+      radius,
+      ...props
+    });
+  }
+
+  initializeState() {
+    const {gl} = this.state;
+    const {attributeManager} = this.state;
+
+    this.setState({
+      model: this.getModel(gl)
+    });
+
+    attributeManager.addInstanced({
+      instancePositions: {size: 4, update: this.calculateInstancePositions},
+      instanceColors: {size: 3, update: this.calculateInstanceColors}
+    });
+  }
+
+  didMount() {
+    this.updateUniforms();
+  }
+
+  willReceiveProps(oldProps, newProps) {
+    super.willReceiveProps(oldProps, newProps);
+    this.updateUniforms();
+  }
+
+  getModel(gl) {
+    const NUM_SEGMENTS = 32;
+    const PI2 = Math.PI * 2;
+
+    let positions = [];
+    for (let i = 0; i < NUM_SEGMENTS; i++) {
+      positions = [
+        ...positions,
+        Math.cos(PI2 * i / NUM_SEGMENTS),
+        Math.sin(PI2 * i / NUM_SEGMENTS),
+        0
+      ];
+    }
+
+    // turn circle into a cone
+    positions = [0, 0, 1, ...positions, 1, 0, 0];
+
+    return new Model({
+      program: new Program(gl, {
+        id: 'voronoi',
+        vs: assembleShader(gl, {vs: glslify('./voronoi-layer-vertex.glsl')}),
+        fs: glslify('./voronoi-layer-fragment.glsl')
+      }),
+      geometry: new Geometry({
+        drawMode: 'TRIANGLE_FAN',
+        positions: new Float32Array(positions)
+      }),
+      isInstanced: true
+    });
+  }
+
+  updateUniforms() {
+    this.setUniforms({
+      radius: this.props.radius
+    });
+  }
+
+  calculateInstancePositions(attribute) {
+    const {data, getPosition, getRadius} = this.props;
+    const {value, size} = attribute;
+    let i = 0;
+    for (const point of data) {
+      const position = getPosition(point);
+      const radius = getRadius(point) || 1;
+      value[i + 0] = position[0] || 0;
+      value[i + 1] = position[1] || 0;
+      value[i + 2] = position[2] || 0;
+      value[i + 3] = radius || 1;
+      i += size;
+    }
+  }
+
+  calculateInstanceColors(attribute) {
+    const {data, getColor} = this.props;
+    const {value, size} = attribute;
+    let i = 0;
+    for (const point of data) {
+      const color = getColor(point);
+      // use random colors for demostration
```

Comment:
It would be possible and desirable to assign one of [four colors][4C] to each veronoi cell. It would involve a dynamic programming solution, traversing the adjacency matrix of these cells, assigning the color not currently used by an adjacent cell.
[4C]: https://en.wikipedia.org/wiki/Four_color_theorem